kotlin/viewshed-geoprocessing/src/main/java/com/esri/arcgisruntime/sample/viewshedgeoprocessing/MainActivity.kt

```diff
@@ -0,0 +1,218 @@
+    inputGraphicsOverlay.graphics.add(Graphic(point))
+  }
+
+  /**
+   * Uses the given point to create a FeatureCollectionTable which is passed to performGeoprocessing.
+   *
+   * @param point in MapView coordinates.
+   */
+  private fun calculateViewshedFrom(point: Point) {
+    // remove previous graphics
+    resultGraphicsOverlay.graphics.clear()
+
+    // cancel any previous job
+    geoprocessingJob?.cancel()
+
+    // create field with same alias as name
+    val field = Field.createString("observer", "", 8)
+
+    // create feature collection table for point geometry
+    val featureCollectionTable =
+      FeatureCollectionTable(listOf(field), GeometryType.POINT, point.spatialReference)
+    featureCollectionTable.loadAsync()
+
+    // create a new feature and assign the geometry
+    val newFeature = featureCollectionTable.createFeature().apply {
+      geometry = point
+    }
+
+    // add newFeature and call perform Geoprocessing on done loading
+    featureCollectionTable.addFeatureAsync(newFeature)
+    featureCollectionTable.addDoneLoadingListener {
+      if (featureCollectionTable.loadStatus == LoadStatus.LOADED) {
+        performGeoprocessing(featureCollectionTable)
+      }
+    }
+  }
+
+  /**
+   * Creates a GeoprocessingJob from the GeoprocessingTask. Displays the resulting viewshed on the map.
+   *
+   * @param featureCollectionTable the feature collection table containing the observation point.
+   */
+  private fun performGeoprocessing(featureCollectionTable: FeatureCollectionTable) {
+
+    // geoprocessing parameters
+    val parameterFuture: ListenableFuture<GeoprocessingParameters> =
+      geoprocessingTask.createDefaultParametersAsync()
+    parameterFuture.addDoneListener {
+      try {
+        val parameters = parameterFuture.get().apply {
+          processSpatialReference = featureCollectionTable.spatialReference
+          outputSpatialReference = featureCollectionTable.spatialReference
+
+          // use the feature collection table to create the required GeoprocessingFeatures input
+          inputs["Input_Observation_Point"] = GeoprocessingFeatures(featureCollectionTable)
+        }
+
+        // initialize job from geoprocessingTask
+        geoprocessingJob = geoprocessingTask.createJob(parameters)
+
+        // start the job
+        geoprocessingJob?.start()
+
+        // listen for job success
+        geoprocessingJob?.addJobDoneListener {
+          if (geoprocessingJob?.status == Job.Status.SUCCEEDED) {
+            // get the viewshed from geoprocessingResult
+            (geoprocessingJob?.result?.outputs?.get("Viewshed_Result") as? GeoprocessingFeatures)?.let { viewshedResult ->
+              // for each feature in the result
+              for (feature in viewshedResult.features) {
+                // add the feature as a graphic
+                resultGraphicsOverlay.graphics.add(Graphic(feature.geometry))
+              }
+            }
+          } else {
+            Toast.makeText(
```

Comment:
let's add an error log in here too:
            `Log.e(TAG, geoprocessingJob?.error?.cause.toString())`